diff -up cde/admin/IntegTools/dbTools/udbParseLib.awk.orig cde/admin/IntegTools/dbTools/udbParseLib.awk
--- cde/admin/IntegTools/dbTools/udbParseLib.awk.orig	2012-08-16 15:53:41.000000000 -0400
+++ cde/admin/IntegTools/dbTools/udbParseLib.awk	2012-08-16 17:22:37.548076214 -0400
@@ -342,11 +342,11 @@ function tokenize() {
 	#
         # allow escaping of significant syntax characters
         #
-        gsub("[\\\][{]","{")
-        gsub("\\\:",":")
-        gsub("\\\;",";")
-        gsub("\\\=","=")
-        gsub("[\\\][}]","}")
+        gsub("[\\][{]","{")
+        gsub("\\:",":")
+        gsub("\\;",";")
+        gsub("\\=","=")
+        gsub("[\\][}]","}")
 
 	#
 	# Having insured that interesting chars are surrounded by blanks
