--- cde-2.3.2/programs/dtappbuilder/src/libABil/bil_loadfile.c-orig	2021-06-13 17:44:38.475987550 +0200
+++ cde-2.3.2/programs/dtappbuilder/src/libABil/bil_loadfile.c	2021-06-13 17:44:46.859907845 +0200
@@ -60,7 +60,7 @@
 #include "bilP.h"
 
 BIL_LOAD_INFO	bilP_load;
-char		Buf[MAXPATHLEN];
+static char		Buf[MAXPATHLEN];
 
 /*
  * Read a BIL file into memory. Returns a project node.
--- cde-2.3.2/programs/dtappbuilder/src/ab/ab_bil.c-orig	2021-06-13 17:39:45.116706922 +0200
+++ cde-2.3.2/programs/dtappbuilder/src/ab/ab_bil.c	2021-06-13 17:39:59.044591192 +0200
@@ -88,7 +88,7 @@
 		ABProfiledInterval	configTime
 		);
 
-char	Buf[MAXPATHLEN];	/* Work buffer */
+static char	Buf[MAXPATHLEN];	/* Work buffer */
 
 /*
  * Loads in a new project, replacing the current project.
--- cde-2.3.2/programs/dtappbuilder/src/ab/cgen_utils.c-orig	2021-06-13 17:37:18.269927162 +0200
+++ cde-2.3.2/programs/dtappbuilder/src/ab/cgen_utils.c	2021-06-13 17:37:52.056646411 +0200
@@ -108,7 +108,7 @@
 } CG_STATUS;
 
 
-CGenOptions		CodeGenOptions;
+extern CGenOptions		CodeGenOptions;
 StringList	    user_env_vars = NULL;
 StringList	    module_list = NULL;
 static CG_GOAL      user_goal = CG_GOAL_UNDEF;
--- cde-2.3.2/programs/dtappbuilder/src/ab/proj.c-orig	2021-06-13 17:42:35.676154984 +0200
+++ cde-2.3.2/programs/dtappbuilder/src/ab/proj.c	2021-06-13 17:42:42.896086344 +0200
@@ -228,7 +228,7 @@
 static XRectangle       *rband_rect = NULL;
 static Boolean          mselect_adjust = False;
 
-char			Buf[MAXPATHLEN];	/* Work buffer */
+static char			Buf[MAXPATHLEN];	/* Work buffer */
 
 /*************************************************************************
 **
--- cde-2.3.2/programs/dtappbuilder/src/ab/projP_utils.c-orig	2021-06-13 17:44:16.691194654 +0200
+++ cde-2.3.2/programs/dtappbuilder/src/ab/projP_utils.c	2021-06-13 17:44:22.583138637 +0200
@@ -93,7 +93,7 @@
 **                                                                      **
 **************************************************************************/
 
-char		Buf[MAXPATHLEN];	/* Work buffer */
+static char		Buf[MAXPATHLEN];	/* Work buffer */
 
 /*************************************************************************
 **                                                                      **
--- cde-2.3.2/programs/dtappbuilder/src/ab/proj_stubs.c-orig	2021-06-13 17:41:05.196015155 +0200
+++ cde-2.3.2/programs/dtappbuilder/src/ab/proj_stubs.c	2021-06-13 17:41:13.361937529 +0200
@@ -86,7 +86,7 @@
  * Declarations of global widgets used by callbacks.
  */
 
-char	Buf[MAXPATHLEN];	/* Work buffer */
+static char	Buf[MAXPATHLEN];	/* Work buffer */
 
 /*
  * End declarations of global widgets
--- cde-2.3.2/programs/dtcalc/motif.h-orig	2021-06-13 17:31:59.283577801 +0200
+++ cde-2.3.2/programs/dtcalc/motif.h	2021-06-13 17:33:00.520068953 +0200
@@ -87,7 +87,7 @@
 
 typedef struct Xobject *XVars ;
 
-XVars X ;
+extern XVars X ;
 
 #ifndef CDE_INSTALLATION_TOP
 #define CDE_INSTALLATION_TOP "/usr/dt"
--- cde-2.3.2/programs/dtcalc/motif.c-orig	2021-06-13 17:31:49.452659485 +0200
+++ cde-2.3.2/programs/dtcalc/motif.c	2021-06-13 17:34:03.855542663 +0200
@@ -114,6 +114,10 @@
 
 Boolean colorSrv;
 
+XVars X ;
+ApplicationArgs application_args;
+struct calcVars CalcVars;
+
 static XtResource resources[] =
 {
    {
--- cde-2.3.2/programs/dtcalc/calctool.h-orig	2021-06-13 17:31:24.363867966 +0200
+++ cde-2.3.2/programs/dtcalc/calctool.h	2021-06-13 17:34:06.126523792 +0200
@@ -552,7 +552,9 @@
 #ifdef hpux
   int keybdID;
 #endif
-} CalcVars ;
+};
+
+extern struct calcVars CalcVars;
 
 typedef struct calcVars *Vars ;
 
@@ -568,7 +570,7 @@
    char *session;
 } ApplicationArgs, *ApplicationArgsPtr;
 
-ApplicationArgs application_args;
+extern ApplicationArgs application_args;
 
 /* MP definitions. */
 
--- cde-2.3.2/programs/dtcreate/dtcreate.h-orig	2021-06-13 17:53:27.889954525 +0200
+++ cde-2.3.2/programs/dtcreate/dtcreate.h	2021-06-13 18:00:11.377253141 +0200
@@ -287,7 +287,7 @@
 */
 STORAGECLASS  Widget       FC_NameOrPathText;
 STORAGECLASS  Widget       FC_AndLabel1;
-STORAGECLASS  Widget       FC_PermissionForm;
+extern Widget       FC_PermissionForm;
 STORAGECLASS  Widget       FC_ReadToggle;
 STORAGECLASS  Widget       FC_WriteToggle;
 STORAGECLASS  Widget       FC_ExecuteToggle;
@@ -324,7 +324,7 @@
 #ifdef  NOEXTERN
 
 STORAGECLASS  enum icon_size_range action_icon_size = None_Selected;
-STORAGECLASS  enum icon_size_range filetype_icon_size = None_Selected;
+extern enum icon_size_range filetype_icon_size;
 STORAGECLASS  int     pidIconEditor = 0;
 STORAGECLASS  Boolean bShowPixmaps = TRUE;
 STORAGECLASS  Widget  IconSelector = (Widget)NULL;
--- cde-2.3.2/programs/dthelp/dthelpview/Main.h-orig	2021-06-13 17:19:49.463583176 +0200
+++ cde-2.3.2/programs/dthelp/dthelpview/Main.h	2021-06-13 17:21:06.840961353 +0200
@@ -66,25 +66,25 @@
 
 
 /* Global Variables Used to maintain our cache list of help dialogs */
-CacheListStruct *pCacheListHead; 
-CacheListStruct *pCacheListTale;  
-int             totalCacheNodes;
+extern CacheListStruct *pCacheListHead; 
+extern CacheListStruct *pCacheListTale;  
+extern int             totalCacheNodes;
 
 /* Global Variables */
-Widget topLevel;
-Widget viewWidget;
-Widget manWidget;
-Widget manBtn;
-Widget manText;
-Widget manForm;
-Widget closeBtn;
+extern Widget topLevel;
+extern Widget viewWidget;
+extern Widget manWidget;
+extern Widget manBtn;
+extern Widget manText;
+extern Widget manForm;
+extern Widget closeBtn;
 
 
 /* General global variables */
-int runMode;
-char      *helpClass;
-Display	  *appDisplay;
-char      *startCommand;
+extern int runMode;
+extern char      *helpClass;
+extern Display	  *appDisplay;
+extern char      *startCommand;
 
 
 
--- cde-2.3.2/programs/dthelp/dthelpview/Main.c-orig	2021-06-13 17:20:02.365479486 +0200
+++ cde-2.3.2/programs/dthelp/dthelpview/Main.c	2021-06-13 17:21:12.511915779 +0200
@@ -80,6 +80,24 @@
 #include "UtilI.h"
 #include "ManPageI.h"
 
+
+CacheListStruct *pCacheListHead; 
+CacheListStruct *pCacheListTale;  
+int             totalCacheNodes;
+
+Widget topLevel;
+Widget viewWidget;
+Widget manWidget;
+Widget manBtn;
+Widget manText;
+Widget manForm;
+Widget closeBtn;
+
+int runMode;
+char      *helpClass;
+Display	  *appDisplay;
+char      *startCommand;
+
 /*  Application resource list definition  */
 
 static XrmOptionDescRec option_list[] =
--- cde-2.3.2/programs/dthelp/dthelpdemo/Main.c-orig	2021-06-13 17:23:14.578934833 +0200
+++ cde-2.3.2/programs/dthelp/dthelpdemo/Main.c	2021-06-13 17:23:42.118709030 +0200
@@ -102,6 +102,9 @@
 
 
 
+CacheListStruct *pCacheListHead;
+CacheListStruct *pCacheListTale;
+int             totalCacheNodes;
 
 /* Global variables for the Main module */
 char   *appName;
--- cde-2.3.2/programs/dthelp/dthelpdemo/Main.h-orig	2021-06-13 17:23:16.523919211 +0200
+++ cde-2.3.2/programs/dthelp/dthelpdemo/Main.h	2021-06-13 17:23:46.004676743 +0200
@@ -64,9 +64,9 @@
 
 /* Global Variables Used by our helpCache */
 
-CacheListStruct *pCacheListHead;
-CacheListStruct *pCacheListTale;
-int             totalCacheNodes;
+extern CacheListStruct *pCacheListHead;
+extern CacheListStruct *pCacheListTale;
+extern int             totalCacheNodes;
 
 /*  Globally referenced widget variables */
 extern Widget topLevel;
--- cde-2.3.2/programs/dticon/process.c-orig	2021-06-13 17:27:24.025865079 +0200
+++ cde-2.3.2/programs/dticon/process.c	2021-06-13 17:28:56.704094963 +0200
@@ -101,13 +101,13 @@
 static void Do_DropOp(void);
 
 extern Widget optionsMenu_grid;
-Widget editMenu_paste_pb;
-Widget editMenu_cut_pb;
-Widget editMenu_copy_pb;
-Widget editMenu_rotate_pb;
-Widget editMenu_flip_pb;
-Widget editMenu_scale_pb;
-Widget editMenu_undo_pb;
+extern Widget editMenu_paste_pb;
+extern Widget editMenu_cut_pb;
+extern Widget editMenu_copy_pb;
+extern Widget editMenu_rotate_pb;
+extern Widget editMenu_flip_pb;
+extern Widget editMenu_scale_pb;
+extern Widget editMenu_undo_pb;
 extern Widget newWidthText, newHeightText;
 
 extern GC scratch_gc;
--- cde-2.3.2/programs/dticon/graphics.c-orig	2021-06-13 17:27:19.314904226 +0200
+++ cde-2.3.2/programs/dticon/graphics.c	2021-06-13 17:29:01.565054569 +0200
@@ -60,11 +60,11 @@
 
 extern GC scratch_gc;
 
-Widget editMenu_cut_pb;
-Widget editMenu_copy_pb;
-Widget editMenu_rotate_pb;
-Widget editMenu_flip_pb;
-Widget editMenu_scale_pb;
+extern Widget editMenu_cut_pb;
+extern Widget editMenu_copy_pb;
+extern Widget editMenu_rotate_pb;
+extern Widget editMenu_flip_pb;
+extern Widget editMenu_scale_pb;
 
 extern void Stop_HotBox(void);
 
--- cde-2.3.2/programs/dticon/utils.c-orig	2021-06-13 17:30:14.131451573 +0200
+++ cde-2.3.2/programs/dticon/utils.c	2021-06-13 17:30:21.899387024 +0200
@@ -138,7 +138,7 @@
 static int jskXerrorDebug();
 static int jskXerrorIODebug();
 
-Widget editMenu_undo_pb;
+extern Widget editMenu_undo_pb;
 
 void GetSessionInfo( void );
 void Set_Gfx_Labels( Boolean );
--- cde-2.3.2/programs/dtimsstart/start.c-orig	2021-06-13 18:17:50.958325558 +0200
+++ cde-2.3.2/programs/dtimsstart/start.c	2021-06-13 18:18:59.164757624 +0200
@@ -32,11 +32,6 @@
 #define _NFILE FOPEN_MAX
 #endif
 
-#if !defined(__linux__) && !defined(CSRG_BASED)
-extern char	*sys_errlist[];
-extern int	sys_nerr;
-#endif
-
     /* local functions */
 static int	check_ims_opt(/* ptr */);
 static char	*find_session_resfile(/* res_type */);
@@ -600,7 +595,7 @@
     pid = fork();
     if (pid == (pid_t) -1) {
 	put_xims_log("fork failed [%s]",
-		(errno <= sys_nerr) ? sys_errlist[errno] : NULL, 0, 0);
+		strerror(errno), 0, 0);
 #ifdef	DEBUG
 	perror("fork");
 #endif
@@ -618,7 +613,7 @@
 	execl(SH_PATH, "sh", "-c", renv->cmdbuf, NULL);
 
 	put_xims_log("%s: exec failed [%s]", SH_PATH,
-		(errno <= sys_nerr) ? sys_errlist[errno] : NULL, 0, 0);
+		strerror(errno), 0, 0);
 	/* perror(SH_PATH); */
 	sleep(1);
 	_exit(1);
--- cde-2.3.2/programs/dtksh/ksh93/src/lib/libast/string/fmterror.c-orig	2021-06-13 17:07:47.423385572 +0200
+++ cde-2.3.2/programs/dtksh/ksh93/src/lib/libast/string/fmterror.c	2021-06-13 17:06:13.723138202 +0200
@@ -97,9 +97,5 @@
 
 char*
 fmterror __PARAM__((int err), (err)) __OTORP__(int err;){
-	static char	msg[28];
-
-	if (err > 0 && err <= sys_nerr) return(sys_errlist[err]);
-	sfsprintf(msg, sizeof(msg), "Error %d", err);
-	return(msg);
+	return strerror(err);
 }
--- cde-2.3.2/lib/DtSearch/raima/dbtype.h-orig	2021-06-13 16:17:59.795647192 +0200
+++ cde-2.3.2/lib/DtSearch/raima/dbtype.h	2021-06-13 16:19:00.786253773 +0200
@@ -405,7 +405,8 @@
 struct sk {
    INT sk_fld;
    CHAR_P sk_val;
-} __SK__;
+};
+extern struct sk __SK__;
 typedef struct { struct sk *ptr; LOCK_DESC } SK_P;
 
 /* node key search path stack entry: one per level per key field */
--- cde-2.3.2/lib/DtSearch/raima/makenew.c-orig	2021-06-13 16:20:47.529565226 +0200
+++ cde-2.3.2/lib/DtSearch/raima/makenew.c	2021-06-13 16:20:54.081522963 +0200
@@ -50,6 +50,7 @@
 #include "vista.h"
 #include "dbtype.h"
 
+struct sk __SK__;
 
 /* Set the value of a key field
 */
--- cde-2.3.2/programs/dtstyle/ColorMain.c-orig	2021-06-13 17:25:33.942779822 +0200
+++ cde-2.3.2/programs/dtstyle/ColorMain.c	2021-06-13 17:25:49.505650504 +0200
@@ -246,7 +246,6 @@
 Widget  deleteButton;
 char    *defaultName;
 Bool    WaitSelection;
-int     NumOfPalettes;
 
 /*+++++++++++++++++++++++++++++++++++++++*/
 /* Internal Variables                    */
--- cde-2.3.2/programs/dtudcfonted/mtfgui.c-orig	2021-06-13 18:03:20.333657903 +0200
+++ cde-2.3.2/programs/dtudcfonted/mtfgui.c	2021-06-13 18:16:00.053249039 +0200
@@ -62,13 +62,30 @@
 
 extern FalCodeRegion CodeArea[16];
 
+
+Widget	editPtnW;
+Widget	cpyPtnW;
+Widget	mngPtnW;
+
+EditChar	edg;
+EditList 	edlist;
+EditPane 	edpane;
+XlibInf 	xl;
+DelInf		dn;
+DelPtnInf 	dl[D_MAX] ;
+DelPtnInf 	dl_glyph[EDLIST_MAX];		 /* add dl_disp 1995.09.20 */
+int		sq_disp[EDLIST_MAX];		 /* add sq_disp 1995.09.20 */
+
+EditPtnInf 	em;
+CopyPtnInf 	cpm;
+
 /********************************************************************
  structure of widgets
  ********************************************************************/
 extern	Widget	toplevel;
-	Widget	editPopW,
-		wgeScro,
-		wgeBulB_edit;
+extern	Widget	editPopW,
+		wgeScro;
+	Widget	wgeBulB_edit;
 static	Widget	wgeStaT_form[EDLIST_MAX],
 		wgeStaT_disp[EDLIST_MAX],
 		wgeStaT_list[EDLIST_MAX],
--- cde-2.3.2/programs/dtudcfonted/xoakufont.h-orig	2021-06-13 18:02:55.860864501 +0200
+++ cde-2.3.2/programs/dtudcfonted/xoakufont.h	2021-06-13 18:13:11.610666125 +0200
@@ -255,23 +255,23 @@
 
  **********************************************************************/
 
-Widget	toplevel;
+extern Widget	toplevel;
 
-Widget	editPtnW;
-Widget	cpyPtnW;
-Widget	mngPtnW;
-
-EditChar	edg;
-EditList 	edlist;
-EditPane 	edpane;
-XlibInf 	xl;
-DelInf		dn;
-DelPtnInf 	dl[D_MAX] ;
-DelPtnInf 	dl_glyph[EDLIST_MAX];		 /* add dl_disp 1995.09.20 */
-int		sq_disp[EDLIST_MAX];		 /* add sq_disp 1995.09.20 */
+extern Widget	editPtnW;
+extern Widget	cpyPtnW;
+extern Widget	mngPtnW;
+
+extern EditChar	edg;
+extern EditList 	edlist;
+extern EditPane 	edpane;
+extern XlibInf 	xl;
+extern DelInf		dn;
+extern DelPtnInf 	dl[] ;
+extern DelPtnInf 	dl_glyph[];		 /* add dl_disp 1995.09.20 */
+extern int		sq_disp[];		 /* add sq_disp 1995.09.20 */
 
-EditPtnInf 	em;
-CopyPtnInf 	cpm;
+extern EditPtnInf 	em;
+extern CopyPtnInf 	cpm;
 
 
 /*
--- cde-2.3.2/programs/dthelp/parser/pass2/htag2/global.h-orig	2021-06-13 18:59:17.313216098 +0200
+++ cde-2.3.2/programs/dthelp/parser/pass2/htag2/global.h	2021-06-13 18:45:23.087185702 +0200
@@ -128,20 +128,20 @@
 /*
  * Names for all the input, output and intermediate files we'll need.
 */
-char *inFileName;
-char *outFileName;
-char *errFileName;
-char *idxFileName;
-char *snbFileName;
-char *sortedIdxFileName;
-char *tossFileName;
-char *vstructFileName;
-char *tempFileName;
-char *compFileName;
-char *compZFileName;
+extern char *inFileName;
+extern char *outFileName;
+extern char *errFileName;
+extern char *idxFileName;
+extern char *snbFileName;
+extern char *sortedIdxFileName;
+extern char *tossFileName;
+extern char *vstructFileName;
+extern char *tempFileName;
+extern char *compFileName;
+extern char *compZFileName;
 
-FILE *inFile;
-FILE *outFile;
+static FILE *inFile;
+static FILE *outFile;
 
 typedef struct _loidsRec *LoidsPtr;
 typedef struct _loidsRec {
--- cde-2.3.2/programs/dthelp/parser/pass2/htag2/sdl.c-orig	2021-06-13 18:59:45.987152302 +0200
+++ cde-2.3.2/programs/dthelp/parser/pass2/htag2/sdl.c	2021-06-13 19:01:14.053956362 +0200
@@ -50,6 +50,17 @@
 
 static char wideCharacterString[] = "wide character string";
 
+char *inFileName;
+char *outFileName;
+char *errFileName;
+char *idxFileName;
+char *snbFileName;
+char *sortedIdxFileName;
+char *tossFileName;
+char *vstructFileName;
+char *tempFileName;
+char *compFileName;
+char *compZFileName;
 
 char *mb_malloc(long size)
 {
--- cde-2.3.2/programs/dthelp/parser/canon1/parser/parser.h-orig	2021-06-13 17:12:03.533327442 +0200
+++ cde-2.3.2/programs/dthelp/parser/canon1/parser/parser.h	2021-06-13 17:13:49.699474276 +0200
@@ -301,7 +301,7 @@
 M_PAREXTERN int m_token ;
 M_PAREXTERN int m_scanval ;
 
-char *user_defined_entities M_PARINIT("USER-DEFINED-ENTITIES");
+extern char *user_defined_entities;
 
 /* Declarations for tentative list of omitted tags when checking for tag
    MINimization */
--- cde-2.3.2/programs/dthelp/parser/canon1/parser/parser.c-orig	2021-06-13 17:12:38.555046001 +0200
+++ cde-2.3.2/programs/dthelp/parser/canon1/parser/parser.c	2021-06-13 17:14:12.046294696 +0200
@@ -37,6 +37,8 @@
 
 static void scanloop(LOGICAL prolog);
 
+char *user_defined_entities = "USER-DEFINED-ENTITIES";
+
 /* Main procedure */
 int main(int argc, char **argv)
 {
--- cde-2.3.2/programs/dthelp/parser/canon1/helptag/global.h-orig	2021-06-13 18:55:50.566288240 +0200
+++ cde-2.3.2/programs/dthelp/parser/canon1/helptag/global.h	2021-06-13 18:57:02.656834890 +0200
@@ -227,20 +227,21 @@
     char   *pclass;
     char   *pssi;
     PBLOCK  pprev;
-} block, *pblock;
-PBLOCK blockinfo INIT(NULL);
+};
+extern struct _block block, *pblock;
+extern PBLOCK blockinfo;
 
-LOGICAL inParText INIT(FALSE);
-LOGICAL inSdlP    INIT(FALSE);
+extern LOGICAL inParText;
+extern LOGICAL inSdlP;
 
 /* Save these ids and reuse them on the current virpage.
  * We need two of each (except for the first) so we can alternate to
  * avoid <form> thinking it is supposed to span them.  The bullet id
  * indexes will flip back and forth between 0 and 1 to pick an id.
 */
-int bulletId[2], looseBulletId[2], firstBulletId, firstLooseBulletId;
-int bulletIdIndex INIT(1);
-int looseBulletIdIndex INIT(1);
+extern int bulletId[], looseBulletId[], firstBulletId, firstLooseBulletId;
+extern int bulletIdIndex;
+extern int looseBulletIdIndex;
 
 /* indent for p, image, ex, vex, based on plain list */
 #define LISTFIRSTINDENT 2
@@ -289,7 +290,7 @@
 EXTERN char helptmpbuf[2] INIT("0");
 
 /* Should we use these BASENAME_LIMIT on the size of file names? */
-LOGICAL usingshortnames INIT(FALSE);
+extern LOGICAL usingshortnames;
 /* Limit on how long the basename can be for a .ht file */
 #define BASENAME_LIMIT 6
 /* limit on how many additional character are allowed before the ext */
--- cde-2.3.2/programs/dthelp/parser/canon1/helptag/help.c-orig	2021-06-13 18:56:28.556049335 +0200
+++ cde-2.3.2/programs/dthelp/parser/canon1/helptag/help.c	2021-06-13 18:56:54.459886437 +0200
@@ -32,6 +32,15 @@
 #include "LocaleXlate.h"
 #include "XlationSvc.h"
 
+PBLOCK blockinfo = NULL;
+LOGICAL inParText = FALSE;
+LOGICAL inSdlP = FALSE;
+int bulletId[2], looseBulletId[2], firstBulletId, firstLooseBulletId;
+int bulletIdIndex = 1;
+int looseBulletIdIndex = 1;
+struct _block block, *pblock;
+LOGICAL usingshortnames = FALSE;
+
 void assert_hometopic_exists(void)
 {
 static const char hometopic[] = "-HOMETOPIC";
--- cde-2.3.2/programs/dthelp/parser/pass1/helptag/global.h-orig	2021-06-13 18:29:31.439496397 +0200
+++ cde-2.3.2/programs/dthelp/parser/pass1/helptag/global.h	2021-06-13 18:32:37.930945942 +0200
@@ -170,17 +170,17 @@
 EXTERN M_WCHAR *imageglinktypep;
 EXTERN M_WCHAR *imagegdescription;
 
-LOGICAL inParText INIT(FALSE);
-LOGICAL inSdlP    INIT(FALSE);
+extern LOGICAL inParText;
+extern LOGICAL inSdlP;
 
 /* Save these ids and reuse them on the current virpage.
  * We need two of each (except for the first) so we can alternate to
  * avoid <form> thinking it is supposed to span them.  The bullet id
  * indexes will flip back and forth between 0 and 1 to pick an id.
 */
-int bulletId[2], looseBulletId[2], firstBulletId, firstLooseBulletId;
-int bulletIdIndex INIT(1);
-int looseBulletIdIndex INIT(1);
+extern int bulletId[], looseBulletId[], firstBulletId, firstLooseBulletId;
+extern int bulletIdIndex;
+extern int looseBulletIdIndex;
 
 /* save the textsize to emit it on the <p> tag; this allows us to have
  * a single ssi= for the "ex" <block> and modify the text size individually 
@@ -213,7 +213,7 @@
 EXTERN char *helpext;
 
 /* Should we use these BASENAME_LIMIT on the size of file names? */
-LOGICAL usingshortnames INIT(FALSE);
+extern LOGICAL usingshortnames;
 /* Limit on how long the basename can be for a .ht file */
 #define BASENAME_LIMIT 6
 /* limit on how many additional character are allowed before the ext */
--- cde-2.3.2/programs/dthelp/parser/pass1/helptag/help.c-orig	2021-06-13 18:52:44.982455322 +0200
+++ cde-2.3.2/programs/dthelp/parser/pass1/helptag/help.c	2021-06-13 18:54:10.006920630 +0200
@@ -32,6 +32,14 @@
 #include "LocaleXlate.h"
 #include "XlationSvc.h"
 
+
+LOGICAL inParText = FALSE;
+LOGICAL inSdlP = FALSE;
+int bulletId[2], looseBulletId[2], firstBulletId, firstLooseBulletId;
+int bulletIdIndex = 1;
+int looseBulletIdIndex = 1;
+LOGICAL usingshortnames = FALSE;
+
 void assert_hometopic_exists(void)
 {
 static const char hometopic[] = "-HOMETOPIC";
